imports/wasi_snapshot_preview1/fs_test.go

```diff
@@ -1171,10 +1189,61 @@ func Test_fdReaddir(t *testing.T) {
 					},
 				}
 			},
-			buf: 0, bufLen: 30, // length is longer than the second entry
-			cookie:          2,  // previous d_next
-			expectedBufused: 30, // length to read exactly one more.
-			expectedMem:     append(dirent2, dirent3[0:4]...),
+			buf: 0, bufLen: 30, // length is longer than the second entry, but not long enough for a header.
+			cookie:          1,  // d_next of first
```

Comment:
I have changed cookie 2 to 1, because when you send in cookie 1, you expect the second entry.